### DISCUSSION TITLE
Need help how to put cheat on xml

### DISCUSSION THREAD

**Discussion by jamex00** [rafaelvcaetano/melonDS-android#1067] [General]
Asking how to put cheat on xml specifically digimon story lost evolution, the database i got seem to have the codes but it won't show on the list, and would just have to type in manually myself but the problem is i have no knowledge on xml and dont know the format to put cheats on the file. any help is very much appreciated, thank you.

**Comment by rafaelvcaetano** [comment]
The format of the file is the following:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<codelist>
    <name>Whatever name you want to give your database</name>
    <game>
        <name>The name of the game</name>
        <gameid>The ID of the game in the format AAAA 00000000</gameid>
        <folder>
            <name>The name of the folder</name>
            <cheat>
                <name>The name of the cheat</name>
                <note>Any extra information you might want to add about this cheat</note>
                <codes>00000000 11111111 22222222 33333333</codes>
            </cheat>
        </folder>
    </game>
</codelist>
```

So, in the `<codes>` section is where you actually put the code you want. If you want to add multiple codes, just add multiple `<cheat>` sections next to the first one, like this:

```xml
<cheat>
    <name>Your first cheat</name>
    <note>Any extra information you might want to add about this cheat</note>
    <codes>00000000 11111111 22222222 33333333</codes>
</cheat>
<cheat>
    <name>Your second cheat</name>
    <note>Any extra information you might want to add about this cheat</note>
    <codes>00000000 11111111 22222222 33333333</codes>
</cheat>
```

**Very important** is the game  ID. If you are playing Digimon Story - Lost Evolution (Japanese version), then the game ID is `BLEJ 06ADAD39`. If not, you will have the find out the game ID of your ROM.

Finally, save the file with the `.xml` extension and you will be able to import it into the emulator.

**Comment by mohammeddahhani** [comment]
Does not work for me..

Which step first ? Disable cheats, Upload the xml before start rom, then activate them when rom's running ?
```html


<?xml version="1.0" encoding="UTF-8"?>
<codelist>
	<name>NDS Cheats</name>
	<game>
		<name>Pokemon Sacred Gold</name>
		<gameid>IPKE 66972395</gameid>
		<folder>
			<name>Main</name>
			<cheat>
				<name>Instant Message</name>
				<codes>12002346 00002100 120202EE 00001C0C 120202F0 00004818 1202031E 0000BD10 02020320 E7E53C2D 1202032E 0000D0DF 1202033A 0000E7F1</codes>
			</cheat>
			<cheat>
				<name>Quick Screen Change</name>
				<codes>1200FA78 00002200 1200FA9C 00002200</codes>
			</cheat>
			<cheat>
				<name>Fast Nurse</name>
				<codes>1206325E 0000E000 1206327A 0000E000 9224BDCC 00002000 1224BDCC 00002001 D2000000 00000000</codes>
			</cheat>
			<cheat>
				<name>Fast HP Bar</name>
				<codes>12265A80 00001C20 12265A8E 00001C20 12081750 00001C21</codes>
			</cheat>
			<cheat>
				<name>Fast Game</name>
				<codes>12000104 00000000 B21D1594 00000000 DA000000 00000004 D3000000 00000000 D6000000 02000100 D3000000 00000000 92000100 00001C1C 12000E1E 000046F0 12000E20 000046F0 12000104 00008080 D2000000 00000000 92000100 0000071C 12000E1E 000046C0 12000E20 000046C0 12000104 00008080 D2000000 00000000 A2000104 00008080 12000E1E 0000F0D0 12000E20 0000E826 D2000000 00000000</codes>
			</cheat>
		</folder>
	</game>
</codelist>
```

**Reply by rafaelvcaetano** [reply]
Import the cheats and enable them from the settings. Then, launch your game, pause it, select the "Cheats" option from the pause menu, and select the cheats that you want to use. If it says that it couldn't find cheats for the current ROM, then the `gameid` you wrote must be wrong. Everything else in the file seems to be OK.

**Reply by Masaigu1** [reply]
i have been having a similar problem, i have double checked the gameid so many times now i dont know why its not working. i select the xml file for importing, and i get the progress bar widget notification, then it goes about 75% and just disappears, and there is no cheat found once i check game.  Im trying to do this for PMD explorers of sky. I know the file is correct cuz it works on desktop as a regular AR code manager XML file
